### ISSUE TITLE
Investigate & Reduce duplicate org.apache.jasper.* classes in JSP bundles 

### ISSUE THREAD

**Issue by volosied** [OpenLiberty/open-liberty#22357]
We will need to investigate differences between these classes and see if we can reduce any duplication. Or, perhaps, pull in from the open source.

com.ibm.ws.jsp.jasper:
https://github.com/OpenLiberty/open-liberty/tree/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.jsp.jasper/src/org/apache/jasper

com.ibm.ws.jsp.2.3:
https://github.com/OpenLiberty/open-liberty/tree/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.jsp.2.3/src/org/apache/jasper

com.ibm.ws.jsp:
https://github.com/OpenLiberty/open-liberty/tree/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.jsp/src/org/apache/jasper

Upcoming Pages 3.1 Bundle: io.openliberty.pages.3.1.internal.factories 
- Will contain copies from the com.ibm.ws.jsp.2.3 bundle

Trouble is that these org.apache.jasper classes have IBM code within them. I will add a comment detailing the the IBM code.

Alternatively, it would be nice to update to the latest code from the open source community (maybe with IBM patches applied)?

**Comment by volosied** [comment]
com.ibm.ws.jsp.2.3:
```
runtime/HttpJspBase.java
64:import com.ibm.ws.jsp.JspCoreException;

runtime/PageContextImpl.java
106:import com.ibm.ws.jsp.Constants;

runtime/JspFactoryImpl.java
83:import com.ibm.ws.jsp.runtime.PageContextPool;
84:import com.ibm.ws.util.WSThreadLocal;

runtime/JspApplicationContextImpl.java
45:import com.ibm.wsspi.webcontainer.WCCustomProperties; // PI31922
 
runtime/JspRuntimeLibrary.java
82:import com.ibm.ws.jsp.JspCoreException;
83:import com.ibm.wsspi.webcontainer.WebContainerConstants;

runtime/BodyContentImpl.java
76:import com.ibm.wsspi.webcontainer.WCCustomProperties; // PK95332

runtime/JcdiWrappedJspApplicationContextImpl.java
18:import com.ibm.ws.jsp.webcontainerext.JSPExtensionFactory;
19:import com.ibm.wsspi.webcontainer.WCCustomProperties; // PM05903

runtime/JspContextWrapper.java
90:import com.ibm.ws.jsp.JspCoreException;

runtime/JspWriterImpl.java
80:import com.ibm.ws.jsp.JSPStrBuffer;
81:import com.ibm.ws.jsp.JSPStrBufferFactory;
82:import com.ibm.wsspi.webcontainer.WCCustomProperties;  //PK90190
```

https://www.ibm.com/support/pages/apar/PI31922 - https://issues.apache.org/bugzilla/show_bug.cgi?id=52998
https://www.ibm.com/support/pages/apar/PK95332 - Buffer looks to be set via JVM property `org.apache.jasper.runtime.BodyContentImpl.LIMIT_BUFFER` upstream 
https://www.ibm.com/support/pages/apar/PM05903
https://www.ibm.com/support/pages/apar/PK90190

**Comment by volosied** [comment]
closed in favor of #22341
